### DISCUSSION TITLE
notepad++ features roadmap

### DISCUSSION THREAD

**Discussion by Albirew** [dail8859/NotepadNext#92] [General]
Hi,
Is there a roadmap of notepad++ features that are either not or not yet implemented?
Also, what about compatibility with n++ modules?
Last, is there any plan to make a cross-platform compatible addon system?
Regards.

**Comment by dail8859** [comment]
Hi @Albirew 

> Is there a roadmap of notepad++ features that are either not or not yet implemented?

No. This has really just been a multi-year side project of mine. There's no roadmap at all. Just as I took an interest in something I'd start hacking on it.

There's many Notepad++ features that would be useful, other I think would be better off left out. There's also alot of room for improvement over Notepad++'s implementation of certain features.

> Also, what about compatibility with n++ modules?

There's no plans to support N++ modules. They are just way too specific to the Win32 API that I don't foresee any possible way to make them compatible.

> is there any plan to make a cross-platform compatible addon system?

Yes and no. I have no actual plans but I've tried to keep things modular enough internally that a addon system could be implemented in the future. Ideally if this project would ever get enough usage then I think a cross-platform addon system would be incredibly useful...which means probably something like an interpreted scripting language rather than natively compiled C++ addons.

**Reply by Albirew** [reply]
From my personnal use, i'd say "drop a folder opens everything inside", "copy RTF to clipboard" (NppExportModule, allow to copy text with syntax coloration included in order to paste it elsewhere), and auto-reindentation ("Indent By Fold", was "TextFX Character" before. reindent full document. may need few options like indent size in space chars and indent method: spaces or tabs)

**Comment by Mageteck** [comment]
How do I get **NotePad++ cursor** to behave like a normal text ecitor, **corsor to right of next character**?  (_Win NotePad, etc._)

Now NotePad++ cursot is 2-spaces left of next chafracter.

**Reply by dail8859** [reply]
First off...this isn't Notepad++

If the cursor isnt in the correct location then I'd suggest posting screenshots, what OS you are using or possibly providing a file that has that behavior.

**Reply by Mageteck** [reply]
"**Black Mesa: Definitive Edition Gameplay Walkthrough Part - 6 No Commentary   |**" this is an example of what NotePad++ displayes.  Note the position of the coursor ( | ).  I should be "**....Part - 6 No Commentary|** "

Likelly something in **Settings, Margine/Border/Edge**.

Sorry, I have not figured how to take a screenshot on my Ubuntu Laptop (_no key for that_).

**Ubuntu 17**.

**Reply by dail8859** [reply]
Unfortunately that's not enough for me to even begin to guess what the problem is. Maybe someone else has encountered this and might be able to provide more detaiks.

**Reply by vfedyay** [reply]
@Mageteck try installing Flameshot - there is deb package for Ubuntu 18 and it will certainly will work on Ubuntu 17.
https://github.com/flameshot-org/flameshot/releases
Also if you use Gnome then just open Activities and try typing Screenshot.

And just to be sure - you are using NotepadNext, not a Notepad++ ?

**Comment by mandyedi** [comment]
Hi @dail8859 ,

>There's many Notepad++ features that would be useful, other I think would be better off left out.

Just a simple list of your ideas (what should stay, what should be left out) would be really helpful so if someone wants to implement something it would give a direction. At least I'm very curious. :)
